### PR TITLE
Add bitmask array

### PR DESCRIPTION
**Pull request by calebzulawski**
A follow up on the somewhat problematic feature from #239 (based on that branch).

### CODE REVIEW

**Comment by workingjubilee** [comment]
Aaaand my condolences.

**Comment by calebzulawski** [comment]
Ok, the problem was just not gating enough on the `generic_const_exprs` feature, should be fine now I think.

File: crates/core_simd/src/masks/full_masks.rs [hunk 1]
```diff
@@ -139,6 +142,74 @@ where
...
+    pub fn to_bitmask_array<const N: usize>(self) -> [u8; N]
+    where
+        super::Mask<T, LANES>: ToBitMaskArray,
+        [(); <super::Mask<T, LANES> as ToBitMaskArray>::BYTES]: Sized,
+    {
+        assert_eq!(<super::Mask<T, LANES> as ToBitMaskArray>::BYTES, N);
+
+        // Safety: N is the correct bitmask size
+        //
+        // The transmute below allows this function to be marked safe, since it will prevent
```
Comment (programmerjake) [inline crates/core_simd/src/masks/full_masks.rs:157]: `transmute_copy` doesn't check size iirc, so the comment should be updated to not state that the transmutation is what's checking size.

**Review by programmerjake — CHANGES_REQUESTED** [review]

File: crates/core_simd/src/masks/full_masks.rs (hunk 1 above)
Comment (calebzulawski) [reply crates/core_simd/src/masks/full_masks.rs]: Yep.  It's asserted above, so I just removed the explanation entirely.

**Review by programmerjake — APPROVED** [review]
lgtm

**Pull request merged**